### PR TITLE
tiproxy: add tiproxyctl commands and http api

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

Add a document to introduce TiProxy API and add some `tiproxyctl` commands.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/17340
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by djshow832 — APPROVED** [review]

File: tiproxy/tiproxy-api.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+The following example gets the TiProxy configuration in JSON format:
+
+```bash
+curl "http://127.0.0.1:3080/api/admin/config/?format=json"
+```
+
+## Set TiProxy configuration
+
+This endpoint is used to modify the TiProxy configuration using the TOML format. Unspecified configuration items will remain unchanged, so you only need to specify the items that you want to modify.
```
Comment (qiancai) [inline tiproxy/tiproxy-api.md:42]: there is a comment for this paragraph in the Chinese PR
https://github.com/pingcap/docs-cn/pull/17340/files#r1619712746

File: tiproxy/tiproxy-api.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+### Request URI
+
+`PUT /api/admin/config/`
+
+### Request body
+
+The TiProxy configuration should be provided in TOML format. For example:
```
Comment (qiancai) [inline tiproxy/tiproxy-api.md:50]:
```suggestion
You need to provide the TiProxy configuration in TOML format. For example:
```

File: tiproxy/tiproxy-api.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+```toml
+[log]
+level='warning'
+```
+
+### Example
+
+The following example sets `log.level` as `"warning"`, while leaving other configuration items unchanged.
```
Comment (qiancai) [inline tiproxy/tiproxy-api.md:59]:
```suggestion
The following example sets `log.level` as `'warning'`, while leaving other configuration items unchanged.
```

File: tiproxy/tiproxy-api.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    The output is as follows:
+
+    ```toml
+    [log]
+    encoder = 'tidb'
+    level = 'info'
+    ```
+
+2. Specify the value of `log.level` in the `test.toml` file and send a `PUT /api/admin/config/` request to update the value of `log.level`:
```
Comment (qiancai) [inline tiproxy/tiproxy-api.md:75]:
```suggestion
2. Specify the value of `log.level` in the `test.toml` file, and then send a `PUT /api/admin/config/` request to update the value of `log.level`:
```

File: tiproxy/tiproxy-api.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+    ```toml
+    [log]
+    encoder = 'tidb'
+    level = 'warning'
+    ```
+
+## Get TiProxy health status
+
+This endpoint is used to get the health status of TiProxy and the checksum of the configuration. When TiProxy is running normally, this endpoint returns the checksum of the configuration. When TiProxy is shutting down, it returns an error.
```
Comment (qiancai) [inline tiproxy/tiproxy-api.md:100]:
```suggestion
This endpoint is used to get the health status of TiProxy and the checksum of the configuration. When TiProxy is running normally, this endpoint returns the checksum of the configuration. When TiProxy is shutting down or offline, it returns an error.
```

File: tiproxy/tiproxy-command-line-flags.md [hunk 6]
```diff
@@ -18,11 +18,29 @@ This section lists the flags of the server program `tiproxy`.
...
-## TiProxy control
+## TiProxy Control
 
-This section lists the flags of the client program `tiproxyctl`.
+This section introduces the syntax, options, and commands of the client program `tiproxyctl`.
 
-### `--log_encoder`
+> **Note:**
+>
+> TiProxy Control is specifically designed for debugging purposes and might not be fully compatible with future capabilities introduced in TiProxy. It's not recommended to include this tool in applications or utilities development to get information.
```
Comment (qiancai) [inline tiproxy/tiproxy-command-line-flags.md:27]:
```suggestion
> TiProxy Control is specifically designed for debugging purposes and might not be fully compatible with future capabilities introduced in TiProxy. It's not recommended to include this tool in application or utility development to get information.
```

File: tiproxy/tiproxy-command-line-flags.md [hunk 7]
```diff
@@ -32,35 +50,67 @@ This section lists the flags of the client program `tiproxyctl`.
...
 
 + Specifies the certificate when dialing to the server.
 + Type: `string`
 + Default: `""`
+
+### Commands
+
+#### `config set`
+
+The `tiproxyctl config set` command reads a TOML-formatted configuration from standard input and sets these configuration items to TiProxy. Unspecified configuration items will remain unchanged, so you only need to specify the items that you want to modify.
```
Comment (qiancai) [inline tiproxy/tiproxy-command-line-flags.md:90]:
```suggestion
The `tiproxyctl config set` command reads a TOML-formatted configuration file from standard input and sets these configuration items to TiProxy. Unspecified configuration items will remain unchanged, so you only need to specify the items that you want to modify.
```

File: tiproxy/tiproxy-command-line-flags.md [hunk 8]
```diff
@@ -32,35 +50,67 @@ This section lists the flags of the client program `tiproxyctl`.
...
+
+The following example sets `log.level` as `"warning"`, while leaving other configuration items unchanged.
```
Comment (qiancai) [inline tiproxy/tiproxy-command-line-flags.md:92]:
```suggestion
The following example sets `log.level` as `'warning'`, while leaving other configuration items unchanged.
```

File: tiproxy/tiproxy-command-line-flags.md [hunk 9]
```diff
@@ -32,35 +50,67 @@ This section lists the flags of the client program `tiproxyctl`.
...
+level = 'warning'
+```
+
+#### `config get`
+
+The `tiproxyctl config get` command is used to get the current TiProxy configuration in TOML format.
+
+#### `health`
+
+The `tiproxyctl health` command is used to get the health status of TiProxy and the checksum of the configuration. When TiProxy is running normally, it returns the checksum of the configuration. When TiProxy is shutting down, it returns an error.
```
Comment (qiancai) [inline tiproxy/tiproxy-command-line-flags.md:110]:
```suggestion
The `tiproxyctl health` command is used to get the health status of TiProxy and the checksum of the configuration. When TiProxy is running normally, it returns the checksum of the configuration. When TiProxy is shutting down or offline, it returns an error.
```

File: tiproxy/tiproxy-command-line-flags.md [hunk 10]
```diff
@@ -32,35 +50,67 @@ This section lists the flags of the client program `tiproxyctl`.
...
+
+The output is as follows:
```
Comment (qiancai) [inline tiproxy/tiproxy-command-line-flags.md:112]:
```suggestion
Example output:
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**